ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java

```diff
@@ -149,69 +149,83 @@ public synchronized void checkOptimizeCheckTasks(long checkInterval) {
         checkTasks = new ScheduledTasks<>(ThreadPool.Type.OPTIMIZE_CHECK);
       }
       checkTasks.checkRunningTask(
-          cachedTables.keySet(),
+          optimizeTables.keySet(),
           () -> 0L,
           () -> checkInterval,
           OptimizeCheckTask::new,
           false);
-      LOG.info("Schedule Optimize Checker finished with {} valid tables", cachedTables.keySet().size());
+      LOG.info("Schedule Optimize Checker finished with {} optimizing tables", optimizeTables.size());
     } catch (Throwable t) {
       LOG.error("unexpected error when checkOptimizeCheckTasks", t);
     }
   }
 
   @Override
   public List<TableIdentifier> listCachedTables() {
-    return new ArrayList<>(cachedTables.keySet());
+    return new ArrayList<>(optimizeTables.keySet());
   }
 
   @Override
   public List<TableIdentifier> refreshAndListTables() {
     LOG.info("refresh tables");
     if (!inited) {
       LOG.info("OptimizeService init not completed, not refresh");
-      return new ArrayList<>(cachedTables.keySet());
+      return new ArrayList<>(optimizeTables.keySet());
     }
+    long startTimestamp = System.currentTimeMillis();
     Set<TableIdentifier> tableIdentifiers =
         new HashSet<>(com.netease.arctic.ams.server.utils.CatalogUtil.loadTablesFromCatalog());
     List<TableIdentifier> toAddTables = tableIdentifiers.stream()
-        .filter(t -> !cachedTables.containsKey(t))
+        .filter(t -> !optimizeTables.containsKey(t))
         .collect(Collectors.toList());
-    List<TableIdentifier> toRemoveTables = cachedTables.keySet().stream()
+    List<TableIdentifier> toRemoveTables = getValidTables().stream()
         .filter(t -> !tableIdentifiers.contains(t))
         .collect(Collectors.toList());
 
     addNewTables(toAddTables);
     clearRemovedTables(toRemoveTables);
 
-    return new ArrayList<>(cachedTables.keySet());
+    // clean tasks when self-optimizing disabled
+    optimizeTables.values().stream()
+        .map(TableOptimizeItem::getArcticTable)
+        .forEach(this::disableSelfOptimizing);
+
```

Comment:
I think this will bring `ConcurrentModificationException` exception, because we removed the entry from `optimizeTables` during `disableSelfOptimizing`.